src/hugolib/site_render.go

```diff
@@ -327,6 +327,41 @@ func (s *Site) renderAssetLinks() error {
 	return s.renderAndWritePage(&s.PathSpec.ProcessingStats.Pages, "assetlinks", targetPath, p, templ)
 }
 
+func (s *Site) renderAppLinks() error {
+
+	p, err := newPageStandalone(&pageMeta{
+		s:    s,
+		kind: kindAppLinks,
+		urlPaths: pagemeta.URLPath{
+			URL: ".well-known/apple-app-site-association",
+		}},
+		output.JSONFormat,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	if !p.render {
+		return nil
+	}
+
+	// only render if both config settings are set
+	if s.Cfg.GetString("appLinkPrefix") == "" || s.Cfg.GetString("appLinkBundle") == "" {
+		return nil
+	}
+
+	targetPath := p.targetPaths().TargetFilename
+
+	if targetPath == "" {
+		return errors.New("failed to create targetPath for App Links")
+	}
+
+	templ := s.lookupLayouts("apple-app-site-association", "_default/apple-app-site-association", "_internal/_default/apple-app-site-association")
```

Comment:
It's convention to use underscores instead of hyphens here. See the other template files in the parent directory. Also, I would suggest we use the .json extension here. This allows anyone working on their own template file to have syntax highlighting in their text editor.

Reply:
This is conversion of how Apple wants the file called. 

> To add the associated domain file to your website, create a file named apple-app-site-association (without an extension).
[source](https://developer.apple.com/documentation/safariservices/supporting_associated_domains)

Does this not determine how the file is rendered? If so, then I have no issue with making the update.

Reply:
No. `lookupLayouts` loads templates from either the use/site level directories, a theme, or an internal template. It's what is used to create the final file but it itself is not the final file. This is the same concept as how the template file for a blog post could be called `single.html` even though that's not going to be the name of the blog post file itself when it is rendered.